### PR TITLE
Added parsing verification

### PR DESCRIPTION
**Pull request by JonahSussman**

### CODE REVIEW

File: kai/pydantic_models.py [hunk 1]
```diff
@@ -55,6 +58,31 @@ def separate_sections(document):
...
+                yield from traverse_tree_for_errors(n)
+
+    yield from traverse_tree_for_errors(node)
+
+
+def has_parse_error(language: str, content: str) -> bool:
+    try:
+        parser = get_parser(language)
+    except Exception:
+        return False
```
Comment (fabianvf) [inline kai/pydantic_models.py:76]: Can we log here so we know if parsing passed because the parser couldn't be found?

**Review by fabianvf — COMMENTED** [review]
One note, looks good though!

File: kai/pydantic_models.py (hunk 1 above)
Comment (JonahSussman) [reply kai/pydantic_models.py:76]: Added!

**Comment by fabianvf** [comment]
@JonahSussman I'm seeing consistent failure still with `Transformers.java` returning a partial result and being counted as a success , here are all the relevant logs i saw:

```
DEBUG - 2024-04-04 14:34:52,630 - [pydantic_models.py:131 - parse_file_solution_content()] - Found single codeblock 
package com.redhat.coolstore.utils;

import com.redhat.coolstore.model.CatalogItemEntity;
import com.redhat.coolstore.model.Order;
import com.redhat.coolstore.model.OrderItem;
import com.redhat.coolstore.model.Product;
import com.redhat.coolstore.model.ShoppingCart;
import java.io.StringReader;
import java.io.StringWriter;
import java.util.ArrayList;
import java.util.List;
import jakarta.json.Json;
import jakarta.json.JsonArray;
import jakarta.json.JsonArrayBuilder;
import jakarta.json.JsonObject;
import jakarta.json.JsonReader;
import jakarta.json.JsonWriter;

import java.util.concurrent.ThreadLocalRandom;
import java.util.logging.Logger;

/**
 * Created by tqvarnst on 2017-03-30.
 */
public class Transformers {

    private static final String[] RANDOM_NAMES = {"Sven Karlsson","Johan Andersson","Karl Svensson","Anders Johansson","Stefan Olson","Martin Ericsson"};
    private static final String[] RANDOM_EMAILS = {"sven@gmail.com","johan@gmail.com","karl@gmail.com","anders@gmail.com","stefan@gmail.com","martin@gmail.com"};

    private static Logger log = Logger.getLogger(Transformers.class.getName());

    public static Product toProduct(CatalogItemEntity entity) {
        Product prod = new Product();
        prod.setItemId(entity.getItemId());
        prod.setName(entity.getName());
        prod.setDesc(entity.getDesc());
        prod.setPrice(entity.getPrice());
        if (entity.getInventory() != null) {
            prod.setLocation(entity.getInventory().getLocation());
            prod.setLink(entity.getInventory().getLink());
            prod.setQuantity(entity.getInventory().getQuantity());
        } else {
            log.warning("Inventory for " + entity.getName() + "[" + entity.getItemId()+ "] unknown and missing");
        }
        return prod;
    }

    public static String shoppingCartToJson(ShoppingCart cart) {
        JsonArrayBuilder cartItems = Json.createArrayBuilder();
        cart.getShoppingCartItemList().forEach(item -> {
            cartItems.add(Json.createObjectBuilder()
                .add("productSku",item.getProduct().getItemId())
                .add("quantity",item.getQuantity())
            );
        });

        int randomNameAndEmailIndex = ThreadLocalRandom.current().nextInt(RANDOM_NAMES.length);

        JsonObject jsonObject = Json.createObjectBuilder()
            .add("orderValue", Double.valueOf(cart.getCartTotal()))
            .add("customerName",RANDOM_NAMES[randomNameAndEmailIndex])
            .add("customerEmail",RANDOM_EMAILS[randomNameAndEmailIndex])
            .add("retailPrice", cart.getShoppingCartItemList().stream().mapToDouble(i -> i.getQuantity()*i.getPrice()).sum())
            .add("discount", Double.valueOf(cart.getCartItemPromoSavings()))
            .add("shippingFee", Double.valueOf(cart.getShippingTotal()))
            .add("shippingDiscount", Double.valueOf(cart.getShippingPromoSavings()))
            .add("items",cartItems) 
            .build();
        StringWriter w = new StringWriter();
        try (JsonWriter writer = Json.createWriter(w)) {
            writer.write(jsonObject);
        }
        return w.toString();
    }

    public static Order jsonToOrder(String json) {
        JsonReader jsonReader = Json.createReader(new StringReader(json));
        JsonObject rootObject = jsonReader.readObject();
        Order order = new Order();
        order.setCustomerName(rootObject.getString("customerName"));
        order.setCustomerEmail(rootObject.getString("customerEmail"));
        order.setOrderValue(rootObject.getJsonNumber("orderValue").doubleValue());
        order.setRetailPrice(rootObject.getJsonNumber("retailPrice").doubleValue());
        order.setDiscount(rootObject.getJsonNumber("dis

/home/fabian/.local/lib/python3.12/site-packages/tree_sitter/__init__.py:36: FutureWarning: Language(path, name) is deprecated. Use Language(ptr, name) instead.
  warn("{} is deprecated. Use {} instead.".format(old, new), FutureWarning)
INFO - 2024-04-04 14:34:52,634 - [server.py:556 - get_incident_solutions_for_file()] - END - completed in '158.7091965675354s:  - App: 'coolstore', File: 'src/main/java/com/redhat/coolstore/utils/Transformers.java' with 6 incidents'
```

**Pull request closed without merging**